scripts/can_bundle.py

```diff
@@ -90,17 +85,23 @@ def fetch_bundle(source: str, bundle_uuid: str, bundle_version: str) -> Bundle:
             log.debug('Searching for %r in catalog %r', source, catalog)
             for plugin_source_spec in plugin.sources:
                 if source_ref.spec.eq_ignoring_prefix(plugin_source_spec):
-                    fqid = SourcedBundleFQIDJSON(source=source_ref.to_json(),
-                                                 uuid=bundle_uuid,
-                                                 version=bundle_version)
-                    fqid = plugin.resolve_bundle(fqid)
+                    fqid = SourcedBundleFQIDJSON(source=source_ref.to_json(), **fqid_args)
+                    fqid = plugin.bundle_fqid_from_json(fqid)
                     bundle = plugin.fetch_bundle(fqid)
                     log.info('Fetched bundle %r version %r from catalog %r.',
                              fqid.uuid, fqid.version, catalog)
                     return bundle
     raise ValueError(f'No repository using source {source!r}')
 
 
+def parse_fqid_fields(args: argparse.Namespace) -> JSON:
```

Comment:
Since `parse_fqid_fields()` is called before `fetch_bundle()`, it would make more sense to have it moved up so it is defined before `fetch_bundle()`.